src/mscorlib/src/System/String.cs

```diff
@@ -1208,6 +1272,42 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             }
             return foundCount;
         }
+
+        private struct SplitSeparatorArray
+        {
+            private readonly bool _single;
```

Comment:
Ah, @ellismg, I see what you mean. Since a separator of `null` or `string.Empty` is treated the same, I can use `string.Empty` as the sentinel value. Thanks for the feedback guys!